### DISCUSSION TITLE
Setting up envoy with Skywalking

### DISCUSSION THREAD

**Discussion by azibhassan** [apache/skywalking#12350] [Q&A]
Hi, 

Environment: 
- Docker 
- skywalking/ui v10.0.1 
- banyandb v0.6.1
- services dont have skywalking agent installed

I wanted to ask if it was possible to send traces to skywalking instance via Envoy. I only came across following [solution](https://github.com/apache/skywalking/blob/master/docs/en/setup/envoy/examples/metrics/envoy-v1.16.yaml) which is related to sending envoy metrics to skywalking backend 

I tried setting up skywalking tracer in envoy but it failed, stating. Seems like envoy was sending http v1 request to grpc port which were getting rejected and I was seeing the following error:
```shell
io.netty.handler.codec.http2.Http2Exception: Unexpected HTTP/1.x request: POST 
``` 
envoy filter was configured like following: 
```shell
    - filters:
      - name: envoy.filters.network.http_connection_manager
        typed_config:
          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
          generate_request_id: true
          codec_type: AUTO
          stat_prefix: ingress_http
          tracing:
            provider:
              name: envoy.tracers.skywalking
              typed_config:
                "@type": type.googleapis.com/envoy.config.trace.v3.SkyWalkingConfig
                grpc_service:
                  envoy_grpc:
                    cluster_name: skywalking
                  timeout: 0.25s
```
Would it be possible to send traces without sw agents to skywalking or zipkin. If there are examples, please do share. That would be really helpful 

Also, while using banyandb, I am seeing following error, which crashes the backend server 
```shell
bayandb        | {"level":"info","module":"ETCD","time":"2024-06-21T01:46:22Z","message":"revoking lease 7587879519294313988"}
bayandb        | {"level":"error","module":"ETCD","error":"etcdserver: requested lease not found","time":"2024-06-21T01:46:22Z","message":"failed to revoke lease 7587879519294313988"}
oap exited with code 137
```
Are you aware what would be the cause of it ?
Please let me know if you need any further skywalking or banyandb configuration

**Comment by wu-sheng** [comment]
About database error, @hanahmily Are you aware of it?

**Comment by wu-sheng** [comment]
We have docs about Istio/Envoy with SkyWalking

- https://istio.io/latest/docs/tasks/observability/distributed-tracing/skywalking/
- https://skywalking.apache.org/blog/how-to-use-skywalking-for-distributed-tracing-in-istio/#environment-description Note, 10.0.1 should have same setup with v9 in the demo.

**Reply by azibhassan** [reply]
Hi Wu-Sheng, please correct me if I am wrong. Isnt istio for kubernetes cluster environment? I am asking about setting up mesh in local docker environment?

**Reply by wu-sheng** [reply]
If you are not running Istio, please follow https://github.com/apache/skywalking/discussions/12350#discussioncomment-9834874

**Comment by wu-sheng** [comment]
https://github.com/apache/skywalking/blob/master/docs/en/setup/envoy/examples/metrics/envoy-v1.16.yaml

What you are referring to, it is a metric service forward, not trace. 

This is SkyWalking envoy tracer doc, https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/skywalking
